tests/unit_tests/test_validate_algorithm_request.py

```diff
@@ -0,0 +1,345 @@
+import json
+from unittest.mock import DEFAULT
+from unittest.mock import patch
+
+import pytest
+
+from mipengine.controller.api.exceptions import BadRequest
+from mipengine.controller.api.exceptions import BadUserInput
+from mipengine.controller.api.validator import validate_algorithm_request
+from tests.unit_tests.common_data_elements import (
+    common_data_elements as mocked_common_data_elements,
+)
+from tests.unit_tests.algorithms_specifications import (
+    algorithms_specifications as mocked_algorithms_specifications,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def mock_cdes():
+    with patch(
+        "mipengine.controller.api.validator.common_data_elements",
+        mocked_common_data_elements,
+    ):
+        yield
+
+
+@pytest.fixture(scope="module", autouse=True)
+def mock_node_catalog():
+    with patch.multiple(
+        "mipengine.common.node_catalog.NodeCatalog",
+        pathology_exists=DEFAULT,
+        dataset_exists=DEFAULT,
+    ) as mock_node_catalog:
+        mock_node_catalog["pathology_exists"].side_effect = (
+            lambda x: True
+            if x
+            in {
+                "test_pathology1",
+                "test_pathology2",
+            }
+            else False
+        )
```

Comment:
1. You should not implement the mocked object's logic inside the mock. If you are forced to do so maybe the function under test need simplification.
2. `side_effect` is for side effects, not return values. There is a `Mock.return_value` but off course it takes fixed values, as a mock should, not functions (see 1).

Reply:
As discussed in the call, with the way the tests are created it's much easier and simpler to do it this way.

In the side_effect documentation it's stated that this can be used for dynamically changing the return values.